diff --git a/kiji-hive-adapter/src/main/java/org/kiji/hive/KijiTableRecordReader.java b/kiji-hive-adapter/src/main/java/org/kiji/hive/KijiTableRecordReader.java
index e63e52c..ea4eba4 100644
--- a/kiji-hive-adapter/src/main/java/org/kiji/hive/KijiTableRecordReader.java
+++ b/kiji-hive-adapter/src/main/java/org/kiji/hive/KijiTableRecordReader.java
@@ -32,6 +32,7 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import org.kiji.hive.utils.KijiDataRequestSerializer;
+import org.kiji.schema.HBaseEntityId;
 import org.kiji.schema.Kiji;
 import org.kiji.schema.KijiDataRequest;
 import org.kiji.schema.KijiDataRequestException;
@@ -81,10 +82,10 @@ public class KijiTableRecordReader implements RecordReader<ImmutableBytesWritabl
 
       KijiScannerOptions scannerOptions = new KijiScannerOptions();
       if (inputSplit.getRegionStartKey().length > 0) {
-        scannerOptions.setStartRow(mKijiTable.getEntityId(inputSplit.getRegionStartKey()));
+        scannerOptions.setStartRow(HBaseEntityId.fromHBaseRowKey(inputSplit.getRegionStartKey()));
       }
       if (inputSplit.getRegionEndKey().length > 0) {
-        scannerOptions.setStopRow(mKijiTable.getEntityId(inputSplit.getRegionEndKey()));
+        scannerOptions.setStopRow(HBaseEntityId.fromHBaseRowKey(inputSplit.getRegionEndKey()));
       }
       mScanner = mKijiTableReader.getScanner(dataRequest, scannerOptions);
       mIterator = mScanner.iterator();
